drake_ros_introspection/include/drake_ros_introspection/predicates/declared_by.h

```diff
@@ -0,0 +1,104 @@
+// Copyright 2022 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <algorithm>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <drake/systems/framework/system_base.h>
+
+namespace drake_ros_introspection {
+namespace predicates {
+
+namespace internal {
+
+template <typename BaseT, typename T>
+bool isinstance(const T& value) {
+  return (dynamic_cast<const BaseT*>(&value) != nullptr);
```

Comment:
Is this the same as `return std::is_base_of<BaseT, T>::value`? If not, I'm surprised there isn't something in `type_traits` for it.